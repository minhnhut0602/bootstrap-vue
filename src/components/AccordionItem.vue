<template>
    <div class="accordion-item">
        <h2 class="accordion-header">
            <button
                class="accordion-button collapsed"
                data-bs-toggle="collapse"
                :data-bs-target="`#${id}`"
            >
                <slot name="header">
                    {{ header }}
                </slot>
            </button>
        </h2>

        <div
            :id="`${id}`"
            class="accordion-collapse collapse"
            :data-bs-parent="parentId ? `#${parentId}` : undefined"
        >
            <div class="accordion-body">
                <slot />
            </div>
        </div>
    </div>
</template>

<script lang="ts" setup>
import {defineProps} from 'vue';
import {idProps} from '@/composables/useId';

defineProps({
    header: {
        type: String,
        default: undefined,
    },
    ...idProps,
    parentId: {
        type: String,
        default: undefined,
    },
});
</script>
