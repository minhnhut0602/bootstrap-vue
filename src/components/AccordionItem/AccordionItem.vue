<template>
    <div class="accordion-item">
        <component
            :is="headerTag" 
            class="accordion-header"
            :class="headerClass"
        >
            <button
                class="accordion-button collapsed"
                data-bs-toggle="collapse"
                :data-bs-target="`#${id}`"
            >
                <slot name="header">
                    {{ header }}
                </slot>
            </button>
        </component>

        <div
            :id="`${id}`"
            class="accordion-collapse collapse"
            :data-bs-parent="parentId ? `#${parentId}` : undefined"
        >
            <div class="accordion-body">
                <slot />
            </div>
        </div>
    </div>
</template>

<script lang="ts">
import {idProps} from '@/composables/useId';
</script>

<script lang="ts" setup>
defineProps({
    header: {
        type: String,
        default: undefined,
    },
    headerClass: {
        type: [String, Array, Object],
        default: undefined,
    },
    headerTag: {
        type: String,
        default: 'h2',
    },
    ...idProps,
    parentId: {
        type: String,
        default: undefined,
    },
});
</script>
