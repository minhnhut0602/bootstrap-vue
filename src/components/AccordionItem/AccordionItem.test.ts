import AccordionItem from '.';
import {shallowMount} from '@vue/test-utils';

describe('template', () => {
    componentRenderTest(AccordionItem, {
        // Force an id to prevent snapshots from failing because of uuid.
        props: {id: 'dont-change-please'},
    });

    componentSlotRenderTest(AccordionItem);

    componentSlotRenderTest(AccordionItem, 'header');

    it('renders header', () => {
        const wrapper = shallowMount(AccordionItem, {
            props: {header: 'find-me'},
        });

        expect(wrapper.text()).toContain('find-me');
    });

    it('adds parent id to collapse', () => {
        const wrapper = shallowMount(AccordionItem, {
            props: {parentId: 'i-am-your-father'},
        });

        expect(wrapper.find('.accordion-collapse').attributes('data-bs-parent')).toBe('#i-am-your-father');
    });
});
