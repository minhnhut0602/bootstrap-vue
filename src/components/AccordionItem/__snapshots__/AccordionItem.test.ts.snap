// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `
"<div class=\\"accordion-item\\">
  <h2 class=\\"accordion-header\\"><button class=\\"accordion-button collapsed\\" data-bs-toggle=\\"collapse\\" data-bs-target=\\"#dont-change-please\\"></button></h2>
  <div id=\\"dont-change-please\\" class=\\"accordion-collapse collapse\\">
    <div class=\\"accordion-body\\"></div>
  </div>
</div>"
`;
